py3status/modules/usbguard.py

```diff
@@ -200,182 +82,94 @@ def post_config_hook(self):
-                    None, "org.usbguard.Devices", signal, None, None, 0, self.py3.update
-                )
-        except Exception:
-            raise Exception(STRING_USBGUARD_DBUS)
-
-        self.thread = Thread(target=self._start_loop)
-        self.thread.daemon = True
-        # self.thread.start()
 
-    def _start_loop(self):
-        pass
+        thread = Thread(target=lambda: GLib.MainLoop().run())
+        thread.daemon = True
+        thread.start()
 
-    def _get_devices(self, device_filter):
-        if device_filter in ["any", "None"]:
-            device_filter = "match"
-
-        devices = self.proxy.listDevices("(s)", device_filter)
-        new_devices = []
-
-        for device_id, device_string in devices:
-            try:
-                device = self.cache_devices[device_string]
-            except KeyError:
-                device = {"id": device_id}
-                for name, regex in self.device_keys:
-                    value = regex.findall(device_string) or None
-                    if value:
-                        value = (
-                            value[0]
-                            .encode("latin-1")
-                            .decode("unicode_escape")
-                            .encode("latin-1")
-                            .decode()
-                        )
-                    device[name] = value
-                self.cache_devices[device_string] = device
-            new_devices.append(device)
-
-        return deepcopy(new_devices)
+    def _get_devices(self):
+        try:
+            raw_devices = self.proxy.listDevices("(s)", "match")
+        except Exception:
+            raise Exception(STRING_USBGUARD_DBUS)
 
-    def _manipulate_devices(self, devices, device_filter):
-        new_device = []
-        format_device_list = None
-        self.device_length = len(devices)
-        for index, device in enumerate(devices, 1):
-            if device_filter == "any":
-                pass
-            elif device_filter == "None":
-                device["policy"] = None
-            elif device["policy"] != device_filter:
-                continue
-            for x in self.init["format_button"]:
-                composite = self.py3.safe_format(getattr(self, x), device)
-                device[x] = self.py3.composite_update(
-                    composite, {"index": "{}/{}".format(device["id"], x.split("_")[-1])}
+        devices = []
+        for device_id, string in raw_devices:
+            device = {"id": device_id}
+            for name, regex in self.keys:
+                value = regex.findall(string) or None
+                if value:
+                    value = value[0]
+                    value = value.encode("latin-1").decode("unicode_escape")
+                    value = value.encode("latin-1").decode("utf-8")
+                device[name] = value
+            if device["policy"] == "block":
+                devices.append(device)
```

Comment:
Can you not use this instead? `raw_devices = self.proxy.listDevices("(s)", "block")`.

Reply:
What happen if users change default policy in usbguard config?

Reply:
Wow, I didn't know this - awesome! Done.

We will parameterize if there is at least one person who wants to use a different policy. It's my understanding however that among usbguard users "block" generally means "temporarily block until decision is taken" and "reject" means "I've decided, don't want this device", so if someone complains, before rushing to implement a parameter I would like to find out why they changed the default policy.

Reply:
IIRC there is a `permanent` flag that you also can enable in Qt or via config to default `block` either temporary or permanent, but if it's rejected permanently, it never can be allowed again until it is removed from the filters (which is the long route)... meaning if users accidentally rejected something, then they might wonder why some device isn't working again weeks or months later.

Reply:
That's right, "permanent" is another beast, it remembers your choice and stores it in a file. We chatted with cyrinux when he was testing usbguard from this branch, it was a conscious decision to remove "permanent" flag and let users handle "permanent" actions via CLI or other tools, because the cost of mistake (pressing wrong button accidentally) is high, and you better be sure what you are doing. We can reconsider in the future if there will be demand for this option of course, but I don't think it's a good idea to add it now.